### DISCUSSION TITLE
Error di Shiny

### DISCUSSION THREAD

**Discussion by smisani** [teamalgoritma/community#252] [Q&A] [answered]
mohon bantuannya, saya mendapatkan error seperti ini:

![image](https://user-images.githubusercontent.com/103245111/182936149-a4433d33-c15e-4932-81b9-a338c8802782.png)

![image](https://user-images.githubusercontent.com/103245111/182936263-318d7cbf-bba6-4883-9622-944d98023c55.png)

saya jadi bingung ini kenapa ya?

**Comment by kevbow** [comment, marked as answer]
Halo @smisani, 

dari error yang muncul diberitahu bahwa terdapat argument `choices` yang tidak diisi. Silakan Anda cari argument tersebut karena tidak terdapat nilai default.

**Reply by smisani** [reply]
Terima kasih.

Hamdallah sudah bisa sekarang. 

Namun saya mengalami kendala lagi, tab dataset saya tidak terbaca padahal saya sudah mengikuti cara-cara yang diajarkan di Algoritma:

![image](https://user-images.githubusercontent.com/103245111/183245377-a4de52cb-afb3-4f39-9ec7-ec69aba920e2.png)
 
sudah beberapa kali saya klik masih tidak dapat diakses. Berikut ini kode yang saya buat:

ui.R:
![image](https://user-images.githubusercontent.com/103245111/183245405-7024770f-42a4-4b40-81cf-cdd0d439e790.png)

server.R:
![image](https://user-images.githubusercontent.com/103245111/183245426-9d3b09e9-c40d-4b4d-8c8d-0fcd8eff4d04.png)

Mohon bantuannya ya. Apabila bisa diadakan mentor sekali lagi, that would be very helpful.

Selain itu, saya memiliki beberapa pertanyaan di bawah ini yang saya harap dapat dibantu dicarikan referensinya:

- bagaimana caranya membuat plot berada tepat ditengah-tengah box pada fluidRow?
- bagaimana caranya membedakan warna marker pada leaflet untuk daerah yang berbeda? Saya sudah coba bahkan membuat kolom baru khusus warna tapi masih tidak bisa.
-  apakah `selectInput` dan `selectizeInput` dianggap sebagai dua jenis input yang berbeda?
- bagaimana cara mengubah numeric menjadi currency menggunakan Rp.

Demikian pertanyaan saya. Mohon bantuannya ya. Terima kasih.

Best,
Sani

**Reply by VictorNugraha** [reply]
Hi Bu Sani @smisani,

Q1: Tidak bisa mengakses bagian menu Dataset?
A1: Untuk akses ke menu Dataset, mungkin bisa dipastikan apakah _key value_ pada _tabName_ sudah sama seperti yang dicantumkan pada bagian _menuItem_?

Q2: Bagaimana caranya membuat plot berada tepat ditengah-tengah box pada fluidRow?
A2: Setau saya ketika kita menempatkan plot pada sebuah box, plot tersebut akan secara otomatis memenuhi sebuah box tersebut. Opsi lainnya yang bisa dicoba ketika kita menginginkan box berada ditengah adlah dengan menggunakan **_column_**. Berikut contoh codenya:

```
fluidRow(
  column(12, align="center",
    plotOutput('plot1')
  )
)
```

Q3: Bagaimana caranya membedakan warna marker pada leaflet untuk daerah yang berbeda?
A3: Untuk melakukan hal tersebut, mungkin bisa mengikuti referensi dari gambar yang saya lampirkan di bawah ini, referensi tersebut merupakan referensi resmi dari [leaftlet](http://rstudio.github.io/leaflet/markers.html) 

![image](https://user-images.githubusercontent.com/28379954/183565117-4cd29605-8d0b-45de-8748-98224d2dfd1c.png)

Q4: Apakah selectInput dan selectizeInput dianggap sebagai dua jenis input yang berbeda?
A4: Kedua inputan tersebut bisa dibilang sebagai 2 input yang berbeda, akan tetapi kalo dari segi tujuannya, bisa dibilang 2 input tersebut memiliki inputan yang sama.

Q5: Bagaimana cara mengubah numeric menjadi currency menggunakan Rp.
A5: Sebenernya untuk proses perubahan numerik menjadi sebuah currency yang memiliki simbol Rp, itu tidak kami sarankan karena akan membuat value numerik tersebut tidak dapat diolah lebih lanjut lagi. Disini kami menyarankan untuk menambahkan jenis currencynya pada judul dari kolom tersebut saja.

Demikian informasi yang dapat kami sampaikan mengenai beberapa pertanyaan yang diajukan. Selamat mencoba dan semoga bermanfaat. :D

**Reply by smisani** [reply]
Terima kasih infonya. 

Untuk jawaban pada Q2 dan Q3 tapinya tidak berhasil ya mas.
untuk Q2, hasilnya plotnya masih belum ke tengah:
![image](https://user-images.githubusercontent.com/103245111/183597165-da3da4e7-d7ac-4325-88e5-f9f82899bc9d.png)

ini codenya:

                > box(width = 6,
                > height = 600, align = "center",
                > h4(tags$b("Tren Pendapatan Asli Daerah (dalam Juta)")),
                > plotlyOutput(outputId = "bar"))


Untuk Q3, cara yang mas tuliskan sudah pernah saya coba dan tidak berhasil.
ini codingnya:
  getColor <- function(sosek_ori_clean) {
    sapply(sosek_ori_clean$Pemda, function(Pemda) {
      ifelse(grepl("Provinsi", sosek_ori_clean$Pemda), "blue", 
             ifelse(grepl("Kab", sosek_ori_clean$Pemda), "yellow", "green"))
    })
  }
  
  icons <- awesomeIcons(
    icon = 'ios-close',
    iconColor = 'black',
    library = 'ion',
    markerColor = getColor(sosek_ori_clean)
  )
  
  output$map <- renderLeaflet({
    
    leaflet(data = sosek_ori_clean) %>% 
      addTiles() %>% 
      addMarkers(~Longitude, ~Latitude, icon = icons, popup = ~label)
  })


Hasil map tapi markernya masih belum berubah. Apakah ada yang salah dari coding saya?

Terima kasih banyak sebelumnya,
Sani
